sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -325,11 +325,11 @@ case class Invoke(
 
   @transient lazy val method = targetObject.dataType match {
     case ObjectType(cls) =>
-      val m = cls.getMethods.find(_.getName == encodedFunctionName)
-      if (m.isEmpty) {
-        sys.error(s"Couldn't find $encodedFunctionName on $cls")
-      } else {
-        m
+      try {
```

Comment:
Can we move the code to a method in the base class, and share it with `StaticInvoke`?

Reply:
There is a bit difference between two `method` from `Invoke` and `StaticInvoke`.  For example, `StaticInvoke` doesn't use encoded function name. Is it safe to share `method` between them?

Reply:
We may backport this change, I'd like make sure we don't introduce regression unintentionally.

Reply:
I see. Let's unify them later in the master branch only. Ideally, these 2 invoke functions should be consistent.